### ISSUE TITLE
Compatibility with Next.js

### ISSUE THREAD

**Issue by irot** [onfido/onfido-sdk-ui#668]
I'm trying to get Onfido to work with my Next.js app but am facing different errors depending on the import approach.

I've created a minimal repo that reproduces this here: [next-onfido-test](https://github.com/irot/next-onfido-test). Install the dependencies and run `npm run dev` for development and `npm run build && npm start` for a production build.

The minimal repo tries three approaches to loading the SDK:

**A. import**
```
import { init } from 'onfido-sdk-ui';
```

This will throw a `ReferenceError: window is not defined` error in both development and production builds.

**B. require**
```
componentDidMount () {
  const Onfido = require('onfido-sdk-ui');
};
```

This will throw an `Unhandled Rejection (TypeError): Object(...) is not a function` error in development, but not production builds.

**C. await import**
```
async componentDidMount () {
  const Onfido = await import('onfido-sdk-ui');
}
```

Same result as B.

~I've not tried loading the library using `<script />` and `<link />` tags and will do so now~ It works if I use `<script />` and `<link />`, but I'd prefer not to have to go this route as it makes the codebase a little harder to maintain.

Any idea where I can start to debug this? Appreciate any help and suggestions.

**Comment by kopijunkie** [comment]
Hi @irot 

For `ReferenceError: window is not defined` errors, this is because of Next.js executing code server-side first. `window` is only available on the client-side and not in the node.js server-side. If you dynamically import the SDK and turn SSR off using `const Onfido = dynamic(import ('onfido-sdk-ui'), {ssr:false});` that should solve that issue and get your app working. I have tried this in your minimal repo and can confirm it works there including the `Unhandled Rejection (TypeError): Object(...) is not a function` error.

I will close this issue for now, but feel free to reopen it if you are still having problems.

**Comment by stefaniacardenas** [comment]
Hi @irot, just following up on this issue. I can confirm that there was a bug on our side when trying to import our SDK in a Next.js project.
The issue has been fixed in release [5.3.0](https://github.com/onfido/onfido-sdk-ui/releases/tag/5.3.0).

Please import the library asynchronously as in the following example. 
```
let Onfido;
async componentDidMount() {
      Onfido = await import('onfido-sdk-ui')
}
Onfido.init()
```
Other types of import will probably break because Next.js executes code server-side first where `window` is not available, as stated in the Next.js documentation (see below).

> Next.js is universal, which means it executes code first server-side, then client-side. The window object is only present client-side, so if you absolutely need to have access to it in some React component, you should put that code in componentDidMount. This lifecycle method will only be executed on the client. You may also want to check if there isn't some alternative universal library which may suit your needs.

I hope this helps.

**Comment by LukeStorry** [comment]
Any update on this, still getting a variety of errors
